### ISSUE TITLE
Support for ALTER SYSTEM Statements

### ISSUE THREAD

**Issue by Magnitus-** [cyrilgdn/terraform-provider-postgresql#179]
Hi,

I'm currently managing our postgres servers with Terraform and something I've yet to configure as code to my satisfaction are the system wide configurations of the postgres server.

Initially, my long term gameplan was to stick a postgres configuration file in etcd and have a side-process on the postgres server regularly poll etcd for changes in that file and on changes, update the file and send a signal to postgres to reload configurations.

However, I stumbled upon this gem today while looking for something unrelated and it seems a lot simpler: https://www.postgresql.org/docs/current/sql-altersystem.html

I think I'll want to integrate the above in my terraform workflow when I have the time for it.

I'm wondering if there is an interest in supporting this in this provider. I could write a separate provider for it down the line, but thematically, it feels like it could belong in this provider.

I was thinking about something like this:

```
postgresql_system {
    configurations = {
      work_mem = "6MB"
      maintenance_work_mem = "256MB"
      ...
    }
    reload = true
}
```

Basically, when creating the resource, the "ALTER SYSTEM SET" statements would get executed on each configuration and on deletion, "ALTER SYSTEM RESET" would be executed on each configuration. If reload is set to true, "pg_reload_conf()" would be called at the end on creation as well.

Thoughts?

**Comment by Magnitus-** [comment]
Actually, I was initially considering adding the functionality to this provider, but as I'm now in the process of integrating an HA setup for postgres with patroni (which appears to allow the modification of many postgres configurations via REST endpoints: https://patroni.readthedocs.io/en/latest/rest_api.html#config-endpoint), I'm now thinking that my energy might be better spent either adapting an existing patroni terraform provider or writing my own.

I'll close this issue for now.
